hashintel/hash
# Fix links in the HASH app backend

## 🌟 What is the purpose of this PR?




This PR replaces the `LinkModel` class with a refactored `LinkEntityModel` class (which now extends the `EntityModel` class). It additionally fixes the following:
- the `ensureSystemEntitiesExists` script that is run in `api/src/index.ts`
- the `seedOrgsAndUsers` script that is run in `api/src/index.ts`
- the `EntityModel` integration tests
- the `LinkEntityModel` integration tests

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203271043333970/1203361844133479/f) _(internal)_

## 🔍 What does this change?




See commits.

## 📜 Does this require a change to the docs?



No.

## ⚠️ Known issues



- `EntityModel.fromEntity` and `LinkEntityModel.fromEntity` accept either an `Entity` or an `EntityModel`, as their types overlap. This is a problem as we don't want to be storing an `EntityModel` on another `LinkModel` or `EntityModel`.

## 🐾 Next steps



## 🛡 What tests cover this?



- `LinkModel` integration tests
- `EntityModel` integration tests

## ❓ How to test this?



1. Checkout the branch / view the deployment
2. Run the external services locally
3. Run `yarn dev` in a separate terminal, and ensure the fixed scripts don't throw any errors

Additionally you can check that the `EntityModel` and `LinkEntityModel` integration tests are working locally using:
- `yarn test:backend-integration /linkEntity.model /entity.model`
